### PR TITLE
Singel product page

### PR DESCRIPTION
**Pull request by mohammadkhizerkhan**
-fixed home category bug
-added single product feature
-added cart to single product

### CODE REVIEW

File: src/backend/db/products.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ import {Logo,games,controller,accessories,xbox_series_x,ps5,ps4,ps3,nintendo,ps5
 export const products = [
   {
     _id: uuid(),
+    id:"Alan_Awake",
```
Comment (znjs) [inline src/backend/db/products.js:12]: Why define id again:??

File: src/App.js [hunk 2]
```diff
@@ -10,6 +10,7 @@ import Mockman from "mockman-js"
 import SignUp from "./components/auth/SignUp";
 import Logout from "./components/auth/Logout";
 import Footer from "./containers/home/Footer";
+import SingleProduct from "./components/single-product/SingleProduct";
```
Comment (ScoobyDubeyDo) [inline src/App.js:13]: **suggestion** : Use berrel export

File: src/components/productpage/productlisting/Product.jsx [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import React from "react";
 import { MdLocalOffer } from 'react-icons/md';
 import { useAuth } from "../../../store/data/AuthContext";
-import { useNavigate } from "react-router-dom";
+import { Link, useNavigate } from "react-router-dom";
 import { useCart } from "../../../store/data/CartContext";
 import { useWishlist } from "../../../store/data/WishlistContext";
```
Comment (znjs) [inline src/components/productpage/productlisting/Product.jsx:6]: try using the index file in the store folder. you can import all in one line

File: src/components/productpage/productlisting/ProductListing.jsx [hunk 4]
```diff
@@ -1,14 +1,15 @@
-import React, { useState } from "react";
+import React, { useState,useEffect } from "react";
 import { Category, FilterHeader, Range, Rating, SortBy } from "../filter";
 import { useData } from "../../../store/data/DataContext";
 import { useFilter } from "../../../store/data/FilterContext";
 import {Product} from "./Product.jsx"
 import { BsFilterLeft } from "react-icons/bs";
+import {ACTION_TYPE} from "../../../store/Actions"
 // import {filterToggleHandler} from "../../../services/utils"
```
Comment (znjs) [inline src/components/productpage/productlisting/ProductListing.jsx:8]: can remove the commented code

File: src/components/single-product/SingleProduct.jsx [hunk 5]
```diff
@@ -0,0 +1,140 @@
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { useNavigate } from "react-router-dom";
+import { useData } from "../../store/data/DataContext";
+import { MdLocalOffer } from "react-icons/md";
+import { useAuth } from "../../store/data/AuthContext";
+import { useCart } from "../../store/data/CartContext";
+import { useWishlist } from "../../store/data/WishlistContext";
```
Comment (znjs) [inline src/components/single-product/SingleProduct.jsx:7]: try using the index file in the store folder. you can import all in one line

File: src/components/single-product/SingleProduct.jsx [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+    img: "",
+    price: "",
+    original_price: "",
+    rating: 0,
+    desc: "",
+  });
+
+  const { title, img, price, original_price, rating, desc, id } = product;
+  // console.log(wishlistState.wishlist)
+  // console.log(cartState.cart)
```
Comment (znjs) [inline src/components/single-product/SingleProduct.jsx]: remove the commented code

**Review by znjs — CHANGES_REQUESTED** [review]

File: src/components/single-product/SingleProduct.jsx [hunk 7]
```diff
@@ -0,0 +1,140 @@
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { useNavigate } from "react-router-dom";
```
Comment (ScoobyDubeyDo) [inline src/components/single-product/SingleProduct.jsx]: **suggestion**: These can be combined in one line

File: src/components/single-product/SingleProduct.jsx (hunk 5 above)
Comment (ScoobyDubeyDo) [inline src/components/single-product/SingleProduct.jsx:7]: suggestion : Use berrel export

File: src/css/cart.css [hunk 8]
```diff
@@ -52,6 +52,7 @@
 }
 
 .img-responsive{
+    max-inline-size: 100%;
     width: 100%;
     /* height: auto; */
```
Comment (ScoobyDubeyDo) [inline src/css/cart.css]: **suggestion** : delete the unused code

**Review by ScoobyDubeyDo — APPROVED** [review]

File: src/components/single-product/SingleProduct.jsx (hunk 5 above)
Comment (mohammadkhizerkhan) [reply src/components/single-product/SingleProduct.jsx:7]: sure

**Review by znjs — APPROVED** [review]

**Pull request merged**